diff -uNr linux-2.4.18/arch/i386/kernel/dmi_scan.c linux-2.4.18-patched/arch/i386/kernel/dmi_scan.c
--- linux-2.4.18/arch/i386/kernel/dmi_scan.c	Mon Feb 25 20:37:53 2002
+++ linux-2.4.18-patched/arch/i386/kernel/dmi_scan.c	Sun Aug 11 12:28:27 2002
@@ -452,6 +452,11 @@
 			MATCH(DMI_BIOS_VERSION, "A04"),
 			MATCH(DMI_BIOS_DATE, "08/24/2000"), NO_MATCH
 			} },
+	{ broken_apm_power, "HP OmniBook XE3 GF", {	/* Handle problems with APM on HP OmniBook XE3 GF */
+		        MATCH(DMI_SYS_VENDOR, "Hewlett-Packard"),
+			MATCH(DMI_PRODUCT_VERSION, "HP OmniBook XE3 GF"),
+			NO_MATCH, NO_MATCH
+			} },
 	{ set_realmode_power_off, "Award Software v4.60 PGMA", {	/* broken PM poweroff bios */
 			MATCH(DMI_BIOS_VENDOR, "Award Software International, Inc."),
 			MATCH(DMI_BIOS_VERSION, "4.60 PGMA"),
